### PR TITLE
Implement the new user settings pages #546

### PR DESCRIPTION
**Pull request by pieer**

### CODE REVIEW

**Comment by mmbotelho** [comment]
Hello Pierre! Here's some minor UI details I noticed:
---

- The component `.help` should not have a `letter-spacing` set - it should be `0`;
Captura de ecrã 2019-05-01, às 15 23 52

---

- The username should have a `@` character behind it, and it should be light gray;
Captura de ecrã 2019-05-01, às 15 25 55
Captura de ecrã 2019-05-01, às 15 25 34

---

- There should be an indication on the side menu of which page the user is on

Captura de ecrã 2019-05-01, às 15 30 34

**Comment by mmbotelho** [comment]
Hello Pierre! 

Here's some minor UI details I noticed, in no particular order:

---

- [x] The component `.help` should not have a `letter-spacing` set - it should be `0`;
Captura de ecrã 2019-05-01, às 15 23 52

---

- [x] The username should have a `@` character behind it, and it should be light gray;
Captura de ecrã 2019-05-01, às 15 25 55
Captura de ecrã 2019-05-01, às 15 25 34

---

- [x] There should be an indication on the side menu of which page the user is on

Captura de ecrã 2019-05-01, às 15 30 26
Captura de ecrã 2019-05-01, às 15 30 34

---

- [x] The input labels on the **Change Password** modal are centered, should be left (probably a bug on desktop and tablet, mobile looks fine);
- [x] No need for placeholder text inside these fields;
Captura de ecrã 2019-05-01, às 15 34 19

---
- [x] The password should be masked (*****) by default. Clicking the 👁 should reveal it. When the password is visible, there should be an [eye-slash](https://fontawesome.com/icons/eye-slash?style=solid) icon, that masks it again;
Captura de ecrã 2019-05-01, às 15 40 51
Captura de ecrã 2019-05-01, às 15 43 01

---

- I really like the transition between the user settings pages, but is it possible to make the page title do the transition as well (instead of just the content)? 
Captura de ecrã 2019-05-01, às 15 55 17

--- 

- The mobile transitions between modals are great! However, the modal headers are not as discussed [on this Figma mockup](https://www.figma.com/file/ZR0DRPhhUuHdJtf7PbGAM6tW/Group-Income?node-id=2278%3A21359). I understand that you are using the default mobile styles for the modals, but for the mobile modal-inside-modal scenario, I believe having the headers as they are on the mockups is more user-friendly. Let me know if this is hard to do, or if you want me to re-think the approach.

File: .Gruntfile.babel.js
Comment (taoeffect) [inline .Gruntfile.babel.js]: Great job on this :+1:

File: frontend/utils/autofocus.js
Comment (taoeffect) [inline frontend/utils/autofocus.js:9]: Great job on placing this informative comment here :+1:

File: .Gruntfile.babel.js
Comment (taoeffect) [reply .Gruntfile.babel.js]: I am curious @pieer, did you try to do this with just the `alias` plugin (like you did for `~`)? Is there a reason we needed a new plugin?

Comment (taoeffect) [reply .Gruntfile.babel.js]: I played around with it and figured out how to do it with just `rollup-plugin-alias`:

```js
        alias({
          resolve: ['.vue', '.js'],
          // https://vuejs.org/v2/guide/installation.html#Standalone-vs-Runtime-only-Build
          vue: path.resolve('./node_modules/vue/dist/vue.common.js'),
          '~': path.resolve('./'),
          '@views': path.resolve('./frontend/views'),
          '@components': path.resolve('./frontend/views/components'),
          '@containers': path.resolve('./frontend/views/containers'),
          '@utils': path.resolve('./frontend/utils'),
          '@assets': path.resolve('./frontend/assets')
        }),
```

The trick was to add `          resolve: ['.vue', '.js'],`

So `rollup-plugin-alias-module-specifiers` does not need to be added.

**Comment by taoeffect** [comment]
One UI issue: while testing in Firefox I noticed that the modal for both Sign Up and Login have their scrollbars visible:

Screen Shot 2019-05-01 at 5 18 04 PM

File: frontend/views/Home.vue
Comment (taoeffect) [inline frontend/views/Home.vue]: This should use the new `@utils/` shortcut right? Also, every import should specify its extension, so in this case `.js` should be added to the end (I know it didn't have that before, but it's something that should be fixed project-wide as we go)

File: frontend/views/components/AppStyles.vue
Comment (taoeffect) [inline frontend/views/components/AppStyles.vue:110]: Good call on moving this to a mixin!

File: frontend/views/components/CreateGroupSteps/GroupInvitees.vue
Comment (taoeffect) [inline frontend/views/components/CreateGroupSteps/GroupInvitees.vue:69]: Nice!

File: frontend/views/components/Modal/ModalMixins.js
Comment (taoeffect) [inline frontend/views/components/Modal/ModalMixins.js]: Hmm, am I right in understanding that passing `null` will remove *all* `keyup` listeners currently registered on window?

That's not what we want to do here though, right? Instead it should only remove the specific listener you registered. I think you can do that by moving the definition of the listener on line 16 up above line 5, and assigning it to a variable, then referencing that variable on lines 16 and here (24).

**Comment by taoeffect** [comment]
Screen Shot 2019-05-03 at 12 09 34 AM

Found another UI glitch (Firefox Developer Edition). See the currency popup

**Comment by taoeffect** [comment]
Screen Shot 2019-05-03 at 12 10 43 AM

And another, misaligned button

**Comment by taoeffect** [comment]
Screen Shot 2019-05-03 at 12 11 42 AM

Two scrollbars

**Comment by taoeffect** [comment]
Screen Shot 2019-05-03 at 12 13 43 AM

Contribution/Income details is totally borked now

**Comment by taoeffect** [comment]
Very strange! I now cannot click on anything in the UI! Clicking does nothing!

Here are the last console lines:

```
[sbp] okTurtles.events/off 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "unload-modal" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "load-modal", "Settings" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "open-modal" ]
main.js:39076:21
[sbp] okTurtles.events/on 
Array [ "close-modal", hide() ]
main.js:39076:21
saveSettings: 
Object { currentGroupId: "21XWnNHh1Tn89KrkHswaqN1rdMcWrLE6nMygfzQv19x6Tt1EBQ", contracts: (3) […], theme: "green" }
main.js:19319:21
[sbp] okTurtles.events/emit 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/off 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "unload-modal" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/off 
Array [ "unload-modal", enableSubmit() ]
main.js:39076:21
triggerEnter main.js:28970:21
transitionAfterEnter main.js:29021:21
triggerEnter main.js:28970:21
transitionAfterEnter main.js:29021:21
triggerLeave main.js:28984:21
[sbp] okTurtles.events/on 
Array [ "close-modal", hide() ]
main.js:39076:21
targetEnter main.js:28998:21
maskLeave main.js:29061:21
transitionAfterEnter main.js:29021:21
[sbp] okTurtles.events/emit 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/off 
Array [ "close-modal" ]
main.js:39076:21
[sbp] okTurtles.events/emit 
Array [ "unload-modal" ]
```

And I'm currently in the Contributions page, after I changed the theme and after I brought up and closed the "Add Income Details" panel.

It seems like I can reproduce the problem: simply clicking the "Add Income Details" panel to bring up the modal, then clicking on the dark grey background to close it, makes it disappear from the page, but also makes it so that I can't click on anything.

Screen Shot 2019-05-03 at 12 19 38 AM


There are two problems here:

1. First, the "Add Income Details" box shouldn't disappear if I immediately close the modal without doing anything
2. Second, it shouldn't make everything unclickable

**Comment by mmbotelho** [comment]
A couple more things:

**Settings sidebar**
- If the user is on the "My account" page, the "My account" menu item should not be clickable and should not have a hover animation;
Captura de ecrã 2019-05-06, às 14 28 21

- Can we remove the on-hover X translation from the letters? I would rather this one be more subtle, just with the background change;
Captura de ecrã 2019-05-06, às 14 28 08


**Change password modal:**
- The default is now masked password (nice!), but the icons are still not making sense. The icons mentioned below are [eye](https://fontawesome.com/icons/eye?style=regular) and [eye-slash](https://fontawesome.com/icons/eye-slash?style=regular):
Captura de ecrã 2019-05-06, às 15 08 58
Captura de ecrã 2019-05-06, às 15 10 46

- The **change password** button should have the following styles:
Captura de ecrã 2019-05-06, às 15 17 50
Captura de ecrã 2019-05-06, às 15 17 38

**Comment by mmbotelho** [comment]
Looking great @pieer . The last thing I noticed is that on tablet and mobile the menu entries SHOULD NOT have a "selected" state as they do in desktop. This can create confusion to users.

Captura de ecrã 2019-05-09, às 17 15 56

There are some other things with modal navigation that are not 100% yet, but I am totally ok with us dealing with that after this is merged and I've had some more time to properly document it. 

Other than that, can't wait for @taoeffect to finish their review so we can move on. Thank you!

**Comment by taoeffect** [comment]
Hmm, @pieer I'm noticing that there's a serious bug in this PR - refreshing the page after being logged in results in either a blank page or being re-routed to the signup/login page.

Here's what the console spits out:

Screen Shot 2019-05-10 at 9 24 59 PM


```
[sbp] okTurtles.events/emit 
Array [ "contractsModified", {…} ]
main.js:39074:21
[sbp] okTurtles.events/emit 
Array [ "contractsModified", {…} ]
main.js:39074:21
lookup failed! main.js:39096:21
[vue-router] uncaught error during route navigation: main.js:12848:53
Invalid object main.js:14729:23
[vue-router] uncaught error during route navigation: main.js:12848:53
Invalid object main.js:14729:23
[sbp] okTurtles.events/on 
Array [ "load-modal", created()
 ]
main.js:39074:21
[sbp] okTurtles.events/on 
Array [ "unload-modal", created()
 ]
main.js:39074:21
[sbp] okTurtles.events/on 
Array [ "logout", startApp()
 ]
main.js:39074:21
[Backend] subscribed 21XWnNN9Uk298HBJ6uVi7XHamFRUt5DaxGKThYoSPvXsg4YZyh: 
Object { type: "success", data: {…} }
main.js:12805:17
[Backend] subscribed 21XWnNWcZ3igBzpfVJweSU2GRaoUnZwogig6HH5wrPFcSbnibC: 
Object { type: "success", data: {…} }
main.js:12805:17
 vue-devtools  Detected Vue v2.6.3 backend.js:1585:13
[sbp] okTurtles.events/emit 
Array [ "contractsModified", {…} ]
main.js:39074:21
[sbp] okTurtles.events/emit 
Array [ "contractsModified", {…} ]
main.js:39074:21
TypeError: parent.getChild(...) is undefined
chunk-7fb1d9cb.js:256:21
[Backend] unsubscribed 21XWnNN9Uk298HBJ6uVi7XHamFRUt5DaxGKThYoSPvXsg4YZyh: 
Object { type: "success", data: {…} }
main.js:12805:17
[Backend] unsubscribed 21XWnNWcZ3igBzpfVJweSU2GRaoUnZwogig6HH5wrPFcSbnibC: 
Object { type: "success", data: {…} }
main.js:12805:17
```

**Review by taoeffect — APPROVED** [review]
OK, since i can no longer reproduce [this issue](https://github.com/okTurtles/group-income-simple/pull/559#issuecomment-491468232) I'm going to approve this PR to help move us along.

**Pull request merged**
